test/e2e/storage-endpoint.integration.test.js

```diff
@@ -0,0 +1,152 @@
+const supertest = require("supertest");
+const { put, remove } = require("../../src/storage/s3");
+const config = require("../../src/storage/config");
+const ropstenDocument = require("../fixtures/certificate.json");
+
+const API_ENDPOINT = "https://api-ropsten.opencerts.io";
+const request = supertest(API_ENDPOINT);
+
+const uuidV4Regex = new RegExp(
+  /^[0-9A-F]{8}-[0-9A-F]{4}-4[0-9A-F]{3}-[89AB][0-9A-F]{3}-[0-9A-F]{12}$/i
+);
+
+const thatIsUploadResponse = {
+  id: expect.stringMatching(uuidV4Regex),
+  key: expect.any(String),
+  type: expect.stringMatching("OPEN-ATTESTATION-TYPE-1")
+};
+
+const thatIsRetrievedDocument = {
+  document: expect.objectContaining({
+    cipherText: expect.any(String),
+    iv: expect.any(String),
+    tag: expect.any(String)
+  })
+};
+
+describe("storage endpoint test", () => {
+  it("should create document when no placeholder object is there", async () => {
+    await request
+      .get("/storage/create")
+      .set("Content-Type", "application/json")
+      .set("Accept", "application/json")
+      .send({
+        document: ropstenDocument
+      })
+      .expect("Content-Type", /json/)
+      .expect(200)
+      .expect(async res => {
+        await remove({ Bucket: config.bucketName, Key: res.body.id });
+        expect(res.body).toEqual(thatIsUploadResponse);
+      });
+  }, 5000);
```

Comment:
why 5000 ?

Reply:
async api test. without it failing timeout.

Reply:
- Extract the value into a variable and add a comment on the var to explain it
- Change all the test to use the variable.
- Why not higher than 5000 to be safe ?

Reply:
5000 is used most of the places in package.